drivers/scsi/ses.c

```diff
@@ -670,10 +670,33 @@ static void ses_match_to_enclosure(struct enclosure_device *edev,
 	if (refresh)
 		ses_enclosure_data_process(edev, edev_sdev, 0);
 
-	if (scsi_is_sas_rphy(sdev->sdev_target->dev.parent))
+	if (scsi_is_sas_rphy(sdev->sdev_target->dev.parent)) {
 		efd.addr = sas_get_address(sdev);
-	else if (scsi_is_ata(sdev))
+	} else if (scsi_is_ata(sdev)) {
 		efd.addr = sdev->host->host_no + 1;
+	} else {
+		const unsigned char *d;
+		const struct scsi_vpd *vpd_pg83;
+		rcu_read_lock();
+		if ((vpd_pg83 = rcu_dereference(sdev->vpd_pg83)) != NULL) {
+			d = vpd_pg83->data + 4;
+			while (d < sdev->vpd_pg83->data + sdev->vpd_pg83->len) {
```

Comment:
```suggestion
			while (d < vpd_pg83->data + vpd_pg83->len) {
```
Aside of being shorter, you should not be allowed to access sdev->vpd_pg83 value around RCU.

Reply:
Thanks, updated!